src/app/models/chart-models/chart.ts

```diff
@@ -2,19 +2,23 @@ import {ChartType} from './chart-type';
 import {GridsterItem} from 'angular-gridster2';
 import {FormatHelper} from '../../utilities/format-helper';
 import {Concept} from '../constraint-models/concept';
+import {ConceptType} from '../constraint-models/concept-type';
 
 export class Chart {
+
   public readonly id: string;
   private _gridsterItem: GridsterItem;
   private _type: ChartType;
   private _variables: Concept[] = [];
+  private _isValid: boolean;
```

Comment:
unfortunately to check whether the chart is valid or not first it needs to be created, so that it can be validated with fractalis - based on what is returned in getChartParameterDescriptions. Currently only the number of variables is checked, because this is the only validation criteria that is returned from fractalis. There is a plan to extend it as a part of [TMT-625](https://jira.thehyve.nl/browse/TMT-625)

Reply:
how about checking if we can create the chart in the `addChart` method inside fractalis.service? based on the selected chart type and selected variables, we should be able to decide whether we can create the chart or not, right?

Reply:
as we discussed, this will be covered by [TMT-627](https://jira.thehyve.nl/browse/TMT-627)